platform/default/include/mbgl/util/default_thread_pool.hpp

```diff
@@ -9,14 +9,21 @@
 
 namespace mbgl {
 
+struct ThreadLifecycle {
+    virtual void onThreadCreated() const {};
```

Comment:
Could you make this pure virtual, not initialise in the constructor as a default argument and check the unique_ptr if a value is set before calling?